qoqo-qryd/tests/integration/experimental_devices.rs

```diff
@@ -0,0 +1,605 @@
+            .unwrap()
+            .contains("Default")
+            .unwrap());
+        assert!(device_mut
+            .call_method0("available_layouts")
+            .unwrap()
+            .contains("Default")
+            .unwrap());
+
+        let current_layout: String = device
+            .call_method0("current_layout")
+            .unwrap()
+            .extract()
+            .unwrap();
+        let current_layout_mut: String = device_mut
+            .call_method0("current_layout")
+            .unwrap()
+            .extract()
+            .unwrap();
+
+        assert_eq!(current_layout, "Default");
+        assert_eq!(current_layout_mut, "Default");
+
+        assert!(device_mut
+            .call_method1("add_layout", ("OtherLayout",))
+            .is_ok());
+
+        assert!(device
+            .call_method1("switch_layout", ("OtherLayout",))
+            .is_ok());
+        assert!(device_mut
+            .call_method1("switch_layout", ("OtherLayout",))
+            .is_ok());
+
+        assert!(device
+            .call_method0("available_layouts")
+            .unwrap()
+            .contains("Default")
+            .unwrap());
+        assert!(device_mut
+            .call_method0("available_layouts")
+            .unwrap()
+            .contains("Default")
+            .unwrap());
+        assert!(device
+            .call_method0("available_layouts")
+            .unwrap()
+            .contains("OtherLayout")
+            .unwrap());
+        assert!(device_mut
+            .call_method0("available_layouts")
+            .unwrap()
+            .contains("OtherLayout")
+            .unwrap());
+    })
+}
+
+/// Test add_qubit_tweezer_mapping function of ExperimentalDeviceWrapper and ExperimentalMutableDeviceWrapper
+#[test]
+fn test_qubit_tweezer_mapping() {
+    // Setup fake preconfigured device
+    let mut exp = ExperimentalDevice::new();
+    exp.set_tweezer_single_qubit_gate_time("PauliX", 1, 0.23, None);
+    let fake_api_device = ExperimentalDeviceWrapper { internal: exp };
+    pyo3::prepare_freethreaded_python();
+    Python::with_gil(|py| {
+        let fake_api_pypyany = fake_api_device.into_py(py);
+        let device_type_mut = py.get_type::<ExperimentalMutableDeviceWrapper>();
+        let device = fake_api_pypyany.as_ref(py);
+        let device_mut = device_type_mut.call0().unwrap();
+
+        device_mut
+            .call_method1("set_tweezer_single_qubit_gate_time", ("PauliX", 1, 0.23))
+            .unwrap();
+
+        assert!(device
+            .call_method1("add_qubit_tweezer_mapping", (0, 1))
+            .is_ok());
+        assert!(device_mut
+            .call_method1("add_qubit_tweezer_mapping", (0, 1))
+            .is_ok());
+        assert!(device
+            .call_method1("add_qubit_tweezer_mapping", (1, 0))
+            .is_err());
+        assert!(device_mut
+            .call_method1("add_qubit_tweezer_mapping", (1, 0))
+            .is_err());
```

Comment:
Why is this supposed to fail here?

Reply:
I have set it up such that if the tweezer is not present in the device, it returns an error.